ios/FluentUI/Tooltip/TooltipViewController.swift

```diff
@@ -9,285 +9,191 @@ import UIKit
+    private static var sourcePointInAnchorView: CGPoint {
+        switch TooltipViewController.arrowDirection {
+        case .up:
+            return CGPoint(x: TooltipViewController.anchorView.frame.width / 2, y: TooltipViewController.anchorView.frame.height)
+        case .down:
+            return CGPoint(x: TooltipViewController.anchorView.frame.width / 2, y: 0)
+        case .right:
+            return CGPoint(x: 0, y: TooltipViewController.anchorView.frame.height / 2)
+        case .left:
+            return CGPoint(x: TooltipViewController.anchorView.frame.width, y: TooltipViewController.anchorView.frame.height / 2)
         }
+    }
 
-        if let perimeterShadow = view.layer.sublayers?[0] {
-            perimeterShadow.frame = view.bounds
-            perimeterShadow.shadowPath = UIBezierPath(roundedRect: view.bounds, cornerRadius: backgroundCornerRadius).cgPath
-            perimeterShadow.shadowColor = UIColor(dynamicColor: shadowInfo.colorTwo).cgColor
-            perimeterShadow.shadowOpacity = 1
-            perimeterShadow.shadowOffset = CGSize(width: shadowInfo.xTwo, height: shadowInfo.yTwo)
-            perimeterShadow.shadowRadius = shadowInfo.blurTwo
-        }
+    private static var sourcePointInWindow: CGPoint {
+        return TooltipViewController.anchorView.convert(sourcePointInAnchorView, to: TooltipViewController.window)
     }
 
-    private static func labelSizeThatFits(_ size: CGSize,
-                                          text: String,
-                                          isAccessibilityContentSize: Bool,
-                                          tokenSet: TooltipTokenSet,
-                                          isMessage: Bool) -> CGSize {
-        let boundingWidth = isAccessibilityContentSize ? size.width : min(tokenSet[.maximumWidth].float - (2 * TooltipTokenSet.paddingHorizontal), size.width)
-        return text.preferredSize(for: UIFont.fluent(tokenSet[isMessage ? .messageLabelTextStyle : .titleLabelTextStyle].fontInfo), width: boundingWidth)
+    private static var boundingRect: CGRect {
+        let screenMargin = TooltipTokenSet.screenMargin
+        return window.bounds.inset(by: window.safeAreaInsets).inset(by: UIEdgeInsets(top: screenMargin,
+                                                                                     left: screenMargin,
+                                                                                     bottom: screenMargin,
+                                                                                     right: screenMargin))
     }
 
-    private func transformForArrowImageView() -> CGAffineTransform {
-        switch positionController.arrowDirection {
+    private let tooltipView: TooltipView
+    private static var anchorView: UIView = UIView()
+    private(set) static var arrowDirection: Tooltip.ArrowDirection = .down
+    private static var preferredArrowDirection: Tooltip.ArrowDirection = .down
+    private static var window: UIView = UIWindow()
```

Comment:
we don't need static windows? the viewcontroller knows its window

Reply:
if it's not static, then it can't be used in the static methods (Error when trying to use `self.view.window`: Instance member 'view' cannot be used on type 'TooltipViewController')

Reply:
right but if the caller is actually the same class you don't need to refer to it as TooltipViewController.arrowPosition() anymore you can just do arrowPosition because it is just helper method

Reply:
Sorry I'm a bit confused...arrowPosition()'s caller isn't the same class. It's only called twice, both in TooltipView.swift, not TooltipViewController.swift. But also, how does that relate to original window comment?

Reply:
any UIView that is present has access to the window. so tooltipview should have a window if needed. the flow has to be changed. rather than tooltipview asking static tooltipviewcontroller what the size should be to then update its own size, once the tooltipviewcontroller is initialized, it knows where the position should be and tell the tooltipview for with the necessary information that the view needs.